Mathlib/Data/Set/Basic.lean

```diff
@@ -2067,10 +2050,64 @@ theorem mem_diff_singleton_empty {t : Set (Set α)} : s ∈ t \ {∅} ↔ s ∈
   mem_diff_singleton.trans <| and_congr_right' nonempty_iff_ne_empty.symm
 #align set.mem_diff_singleton_empty Set.mem_diff_singleton_empty
 
+theorem subset_insert_iff {s t : Set α} {x : α} :
+    s ⊆ insert x t ↔ s ⊆ t ∨ (x ∈ s ∧ s \ {x} ⊆ t) := by
+  rw [← diff_singleton_subset_iff]
+  by_cases hx : x ∈ s
+  · rw [and_iff_right hx, or_iff_right_of_imp (fun h ↦ (diff_subset _ _).trans h)]
+  rw [diff_singleton_eq_self hx, or_iff_left_of_imp And.right]
+
 theorem union_eq_diff_union_diff_union_inter (s t : Set α) : s ∪ t = s \ t ∪ t \ s ∪ s ∩ t :=
   sup_eq_sdiff_sup_sdiff_sup_inf
 #align set.union_eq_diff_union_diff_union_inter Set.union_eq_diff_union_diff_union_inter
 
+/-! ### Lemmas about pairs -/
+
+--Porting note (#10618): removed `simp` attribute because `simp` can prove it
+theorem pair_eq_singleton (a : α) : ({a, a} : Set α) = {a} :=
+  union_self _
+#align set.pair_eq_singleton Set.pair_eq_singleton
+
+theorem pair_comm (a b : α) : ({a, b} : Set α) = {b, a} :=
+  union_comm _ _
+#align set.pair_comm Set.pair_comm
+
+theorem pair_eq_pair_iff {x y z w : α} :
+    ({x, y} : Set α) = {z, w} ↔ x = z ∧ y = w ∨ x = w ∧ y = z := by
+  simp [subset_antisymm_iff, insert_subset_iff]; aesop
+#align set.pair_eq_pair_iff Set.pair_eq_pair_iff
+
+theorem pair_diff_left (hne : a ≠ b) : ({a, b} : Set α) \ {a} = {b} := by
+  rw [insert_diff_of_mem _ (mem_singleton a), diff_singleton_eq_self (by simpa)]
+
+theorem pair_diff_right (hne : a ≠ b) : ({a, b} : Set α) \ {b} = {a} := by
+  rw [pair_comm, pair_diff_left hne.symm]
+
+@[simp] theorem pair_subset_iff : {a, b} ⊆ s ↔ a ∈ s ∧ b ∈ s := by
```

Comment:
I'm wondering why `Set.insert_subset_iff` isn't a `@[simp]` lemma already. If it were, then this lemma would be proved `by simp`.